rust/diesel/src/order.rs

```diff
@@ -129,15 +148,15 @@ impl Order {
             });
 
         if order_total.is_zero() {
-            return None;
+            return Err("Order contains no products".to_owned());
         }
 
         let new_order = NewOrder {
             user_id: order.user_id,
             order_total,
         };
 
-        Some((new_order, product_units))
+        Ok((new_order, product_units))
     }
 
     fn create_order_lines(order: &Order, product_units: Vec<(Product, i16)>) -> Vec<NewOrderLine> {
```

Comment:
I can't leave a comment on it, but the `delete` function below should also return a result, in this case it would be `Result<(), ErrorType>`.

Reply:
I went for the `Result<bool, String>` as I want distinction between "there is no entity with such id" and "error occurred" situations